### PR TITLE
[CL Message Audit] MsgSwapExactAmountOut

### PR DESCRIPTION
**Pull request by stackman27**
Clickup: (https://app.clickup.com/37420681/v/dc/13nzm9-19747/13nzm9-36787)

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

*(E.g.: This pull request improves documentation of area A by adding ....*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/swaps.go
Comment (stackman27) [inline x/concentrated-liquidity/swaps.go:499]: why not ceil `amountSpecifiedRemaining` in favor of the pool?

File: x/concentrated-liquidity/client/cli/tx.go
Comment (mattverse) [inline x/concentrated-liquidity/client/cli/tx.go:54]: good catch! nice!

File: x/concentrated-liquidity/swaps.go
Comment (mattverse) [inline x/concentrated-liquidity/swaps.go:471]:
```suggestion
```

File: x/concentrated-liquidity/swaps.go
Comment (mattverse) [inline x/concentrated-liquidity/swaps.go:508]: Can we change to use `InvalidTickError` for consistency with other code?

File: x/poolmanager/router.go
Comment (mattverse) [inline x/poolmanager/router.go:295]: I do not know the answer for this :(  Once you find out, would you be able to add the reason we defer to the comment as well?

File: x/poolmanager/router_test.go
Comment (mattverse) [inline x/poolmanager/router_test.go:2487]: can we add one line comments to how we are getting numbers?

**Review by mattverse — APPROVED** [review]
LGTM once defer is figured out

File: x/concentrated-liquidity/swaps.go
Comment (czarcas7ic) [reply x/concentrated-liquidity/swaps.go:508]: Yeah, I also don't think we would be using the nextTick value here

File: x/concentrated-liquidity/swaps.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps.go:514]: the sqrtPriceNextTick response here would not be useful since it would just be returning the nil value. What is useful for the caller to know is the nextTick that was used that produced this error in the first place.

File: x/concentrated-liquidity/types/errors.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/types/errors.go:833]: the message is right but the value should be nextTick

File: x/concentrated-liquidity/types/errors.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/types/errors.go:825]: This doesn't need a custom error. This can just be "there are no more ticks initialized to fill the swap" since there is no extra custom data needed to be known

File: x/poolmanager/router.go
Comment (czarcas7ic) [inline x/poolmanager/router.go:281]: This is the wrong godoc, its talking about in logic when it should be talking about out logic

File: x/poolmanager/router.go
Comment (czarcas7ic) [reply x/poolmanager/router.go:295]: Yes, I believe this should also happen in RouteExactAmountIn as well (we should also add panic recovering tests like what was done here https://github.com/osmosis-labs/osmosis/pull/4248)

File: x/poolmanager/router.go
Comment (czarcas7ic) [inline x/poolmanager/router.go:310]:
```suggestion
	// total_swap_fee = max(swapfee1, swapfee2)
```

File: x/poolmanager/router_test.go
Comment (czarcas7ic) [reply x/poolmanager/router_test.go:2487]: Would also like explanations of where these numbers come from.

File: x/poolmanager/router.go
Comment (stackman27) [reply x/poolmanager/router.go:295]: @mattverse we discussed this in our CL call today, and we removed the defer from `RouteExactAmountIn` for protorev, we came to a conclusion that we should have it here, unless it affects any other part of the code

File: x/poolmanager/router_test.go
Comment (stackman27) [reply x/poolmanager/router_test.go]: added!

File: x/poolmanager/router_test.go
Comment (p0mvn) [inline x/poolmanager/router_test.go:2576]:
```suggestion
```

File: x/poolmanager/router.go
Comment (p0mvn) [inline x/poolmanager/router.go:368]: I think the old one is correct because inactive pool also means that LP logic is disabled
```suggestion
		// check if pool is active, if not error
```

**Review by p0mvn — APPROVED** [review]
LGTM. Nice work. Merging once CI passes

**Pull request merged**
